code/game/objects/items/weapons/stock_parts.dm

```diff
@@ -32,6 +31,34 @@
 	origin_tech = "powerstorage=1"
 	starting_materials = list(MAT_IRON = 50, MAT_GLASS = 50)
 	w_type = RECYK_ELECTRONIC
+	var/stored_charge = 0
+	var/maximum_charge = null
+
+/obj/item/weapon/stock_parts/capacitor/examine(mob/user)
+	..()
+	if(stored_charge)
+		to_chat(user, "<span class='notice'>\The [src.name] is charged to [stored_charge]W.</span>")
+		if(stored_charge == maximum_charge)
+			to_chat(user, "<span class='info'>\The [src.name] has maximum charge!</span>")
+
+/obj/item/weapon/stock_parts/capacitor/attackby(obj/item/weapon/W, mob/user)
+	if(istype(W, /obj/item/weapon/wrench))
+		if(src.loc == user)
+			to_chat(user, "<span class='warning'>\The [src] needs to be on the ground to be secured.</span>")
+			return
+		if(!istype(src.loc, /turf/simulated/floor)) //Prevent from anchoring this to shuttles / space
+			to_chat(user, "<span class='notice'>You can't secure \the [src] to [istype(src.loc,/turf/space) ? "space" : "this"]!</span>")
+			return
+		to_chat(user, "You discharge \the [src] and secure it to the floor.")
+		playsound(get_turf(src), 'sound/items/Ratchet.ogg', 50, 1)
+		switch(name)
```

Comment:
Make this a const variable on the object and use that to determine the type to spawn.